command/harness/delegate/delegate.go

```diff
@@ -118,6 +95,7 @@ func (c *delegateCommand) run(*kingpin.ParseContext) error {
 	if err != nil {
 		return err
 	}
+	defer harness.Cleanup(&c.env, c.poolManager)
```

Comment:
We can retain using a separate go routine for cleanup instead of defer. Defer will run when the main has exited. I have seen sometimes the exit gets stuck and program never exits. In that case, cleanup will never get called.

Reply:
Done. I kept the defer as well since for example if there are any VMs left over after `SetupPool` and it returns in an error, then we still need to delete the VMs.

Reply:
Move cleanup to L96 in that case.

Reply:
Done